### ISSUE TITLE
TTS/STT Use cases

### ISSUE THREAD

**Issue by guest271314** [httpslocal/usecases#43]
**Text-To-Speech (TTS)** 

While the Web Speech API does provide basic capability to communicate with the local speech dispatcher socket [`speechd`](https://github.com/brailcom/speechd) some functionality is not available, including

- SSML input
- File input
- File output

**Speech-To-Text (STT)**

Currently only Chrome, Chromium browsers implement STT. That functionality is based on recording the user voice (or other audio output) locally and sending that recording to a remote server, therefore, also some functionality is not available, including

- Local STT processing
- File input
- File output

**Current workarounds**

- https://github.com/guest271314/SpeechSynthesisRecorder (record audio output of `speechSynthesis.speak()` to get a file output)
- https://github.com/guest271314/SpeechSynthesisSSMLParser (parse SSML before executing `speechSynthesis.speak()` where `espeak` and `espeak-ng` accept SSML input directly)

**TL;DR**

- [How to programmatically send a unix socket command to a system server autospawned by browser or convert JavaScript to C++ souce code for Chromium?](https://stackoverflow.com/questions/48219981/how-to-programmatically-send-a-unix-socket-command-to-a-system-server-autospawne)
- https://bugzilla.mozilla.org/show_bug.cgi?id=1425523
- https://bugs.chromium.org/p/chromium/issues/detail?id=795371
- https://github.com/w3c/speech-api/issues/10
- https://github.com/w3c/speech-api/issues/56
- https://github.com/mozilla/DeepSpeech/issues/2401

Direct communication with one or more **local** TTS/STT applications and control of local audio input and output devices over HTTPS/WSS are the use cases descibed in this issue.

Can file a PR for this if necessary.
